draft-normington-jsonpath-latest.xml

```diff
@@ -202,24 +216,34 @@ root-selector = %x24               ; $ selects document root node
                   <name>Syntax</name>
                   <t>A dot child matcher has a key known as a dot child name or a single asterisk (<tt>*</tt>).
                   </t>
-                  <t>A dot child name corresponds to a key in a JSON object, without the surrounding double quotes (<tt>"</tt>).
+                  <t>A dot child name corresponds to a name in a JSON object.
                   </t>
                   <sourcecode type="abnf">
 matcher = dot-child             ; see below for alternatives
 dot-child = %x2E dot-child-name / ; .&lt;dot-child-name&gt;
             %x2E %x2A             ; .*
-dot-child-name = TBD
+dot-child-name = 1*(
+                   %x2D / ; -
+                   %x30-39 / ; 0-9 (core ABNF: DIGIT)
+                   %x41-5A / %x61-7A / ; A-Z / a-z (core ABNF: ALPHA)
+                   %x5F / ; _
+                   %x80-10FFFF ; any non-ASCII Unicode character
+                 )
                   </sourcecode>
-                  <!-- TODO: define dot-child-name using an identifier like syntax. See https://github.com/cburgmer/json-path-comparison/issues/42 -->
                   <t>More general child names, such as the empty string, are supported by "Union Child" (<xref target="unionchild" format="default"/>).
                   </t>
+                  <t>Note that <tt>dot-child-name</tt> rule follows the philosophy of JSON strings when working with non-ASCII range of Unicode: it's
+                     allowed to contain bit sequences that cannot encode Unicode character (an single unpaired UTF-16 surrogate, for example). It can lead
+                     to unpredictable behavior of software that receives such query.
```

Comment:
This text is similar to that in [RFC 7159 section 8.2](https://tools.ietf.org/html/rfc7159#section-8.2), but our goal is slightly different. RFC 7159 defined the JSON data format whereas we are defining a standard for JSONPath implementations. So we should probably say something about implementations rather than the more general "software". How about the following (which also has some editorial improvements)?
```suggestion
                  <t>Note that the <tt>dot-child-name</tt> rule follows the philosophy of JSON strings and is
                     allowed to contain bit sequences that cannot encode Unicode characters (a single unpaired UTF-16 surrogate, for example).
                     The behaviour of an implementation is undefined for child names which do not encode Unicode characters.
```